packages/block-editor/src/components/keyboard-shortcuts/index.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * External dependencies
+ */
+import { first, last } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { useRef, useEffect, useCallback } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useShortcut } from '@wordpress/keyboard-shortcuts';
+import { __ } from '@wordpress/i18n';
+
+function KeyboardShortcuts() {
+	const target = useRef( null );
+
+	// Registering the shortcuts
+	const { registerShortcut } = useDispatch( 'core/keyboard-shortcuts' );
+	useEffect( () => {
+		registerShortcut(
```

Comment:
I think registration belongs to the initialization of a package. In this case, the block-editor package does both at the same time but I agree that we should be careful about that.

I do think it should always be tied to a component though and `registerBlock` is the wrong approach here (can't change now) to avoid globals. See #8981 for the problems of the globals approach.

Reply:
I might be wrong, but I think a similar issue exists today with the `KeyboardShortcuts` component anyway. If I remember correctly that's why there was a separate `BlockActions` component in the first place—the shortcuts have to be rendered in a different part of the render tree. So it might be a moot point, if the component isn't rendered the shortcuts won't work.

There's one case where this is still a little problematic, which is if you load the editor in Code Editor mode and then open the modal, Code Editor elminates a huge part of the render tree, including the shortcut registration.

Reply:
> There's one case where this is still a little problematic, which is if you load the editor in Code Editor mode and then open the help modal, Code Editor elminates a huge part of the render tree, including the shortcut registration. Most of the Block shortcuts are then not visible.

That's a good point, I'll fix.

Reply:
I split the registration and the handling into two components.